user-service/src/main/java/rs/edu/raf/banka1/services/implementations/OrderServiceImpl.java

```diff
@@ -83,52 +100,307 @@ public void startOrder(final Long orderId) {
-//        }
-//
-//        final ListingBaseDto listingBaseDto = marketService.getStock(marketOrder.getStockId());
-//
-//
-//        Random random = new Random();
-//        Long processedNumber = random.nextLong(marketOrder.getContractSize() - marketOrder.getProcessedNumber()) + 1;
-//        marketOrder.setProcessedNumber(marketOrder.getProcessedNumber() + processedNumber);
-//        if(marketOrder.getContractSize().equals(marketOrder.getProcessedNumber())) {
-//            marketOrder.setStatus(OrderStatus.DONE);
-//            orderRepository.save(marketOrder);
-//            return;
-//        }
-//        orderRepository.save(marketOrder);
-//
-//        final Long volume = Long.valueOf(listingBaseDto.getVolume());
-//        Long remainingQuantity = marketOrder.getContractSize() - marketOrder.getProcessedNumber();
-//
-//        long timeInterval = random.nextLong(24*60/(volume/remainingQuantity));
-//        timeInterval = workingHours.equals(WorkingHoursStatus.AFTER_HOURS) ? timeInterval + 30*60 : timeInterval;
-//
-//        executorService.schedule(() -> processOrder(orderId, workingHours), timeInterval, TimeUnit.SECONDS);
-//    }
+    public boolean changeStatus(final Long id, final OrderStatus orderStatus) {
+        Optional<MarketOrder> optOrder = this.orderRepository.findById(id);
+        if (optOrder.isEmpty()) return false;
+        MarketOrder order = optOrder.get();
+        order.setStatus(orderStatus);
+        order.setLastModifiedDate(System.currentTimeMillis() / 1000);
+        this.orderRepository.save(order);
+        return true;
+    }
+
+    private Double calculatePriceForLimitOrder(OrderType orderType, Long contractSize, Double limitValue, Double stockPrice) {
+        if(orderType.equals(OrderType.BUY)) {
+            return contractSize * Math.min(stockPrice, limitValue); // high(ask) umesto stockPrice
+        } else {
+            return contractSize * Math.max(stockPrice, limitValue); // low(bid) umesto stockPrice
+        }
+    }
+    @Override
+    public boolean createLimitOrder(CreateOrderRequest request) {
+        MarketOrder marketOrder = orderMapper.requestToMarketOrder(request);
+        ListingBaseDto listingBaseDto = marketService.getStock(request.getStockId());
+
+        marketOrder.setPrice(calculatePriceForLimitOrder(
+                marketOrder.getOrderType(),
+                marketOrder.getContractSize(),
+                marketOrder.getLimitValue(),
+                listingBaseDto.getPrice()));
+        marketOrder.setFee(calculateFee(request.getLimitValue(), marketOrder.getPrice()));
+        if (!orderRequiresApprove()) {
+            marketOrder.setStatus(OrderStatus.APPROVED);
+        } else {
+            marketOrder.setStatus(OrderStatus.PROCESSING);
+        }
+        marketOrder = orderRepository.save(marketOrder);
+        startLimitOrder(marketOrder.getId());
+        return true;
+    }
+
+    @Override
+    public void startLimitOrder(Long orderId) {
```

Comment:
Da li ovo radi nakon merge-a?

Reply:
@cvele0 @jsimc @andrejadikiccake

Reply:
Ne znam, to su Jelenine metode, nisam testirao, treba sacekati nju.